weaviate/util.py

```diff
@@ -109,19 +114,42 @@ def file_encoder_b64(file_or_file_path: Union[str, BufferedReader]) -> str:
         If the argument is of a wrong data type.
     """
 
+    def _chunks(buffer: io.BufferedReader, chunk_size: int) -> Generator[bytes, Any, Any]:
+        while True:
+            data = buffer.read(chunk_size)
+            if not data:
+                break
+            yield data
+
+    should_close_file = False
+
     if isinstance(file_or_file_path, str):
         if not os.path.isfile(file_or_file_path):
             raise ValueError("No file found at location " + file_or_file_path)
-        with open(file_or_file_path, "br") as file:
-            content = file.read()
-
-    elif isinstance(file_or_file_path, BufferedReader):
-        content = file_or_file_path.read()
+        file = open(file_or_file_path, "br")
+        should_close_file = True
+    elif isinstance(file_or_file_path, Path):
+        if not file_or_file_path.is_file():
+            raise ValueError("No file found at location " + str(file_or_file_path))
+        file = file_or_file_path.open("br")
+        should_close_file = True
+    elif isinstance(file_or_file_path, io.BufferedReader):
+        file = file_or_file_path
     else:
         raise TypeError(
             '"file_or_file_path" should be a file path or a binary read file' " (io.BufferedReader)"
         )
-    return base64.b64encode(content).decode("utf-8")
+
+    if use_buffering:
```

Comment:
Are there any reasons to decide against buffering?

Reply:
When the file is small enough that the overhead of chunking and looping makes it slower than reading in one go. But I have no good idea of a benchmark for when this might occur so we could have `use_buffering=True` instead

Reply:
how about just doing `use_buffering = len(file)> XXX` and remove that option?

Reply:
Ooo nice idea!

Reply:
Hmm, perhaps not. Reading the length of `BufferedReader` is `O(n)` so it would only be useful if the user supplies a path, in which case we'd use `os.path.getsize()` which is `O(1)`. Do you think `use_buffering` will be confusing for some?

Reply:
I think people will miss it. But it is not too important - let's leave it like it is

Reply:
Perhaps check the size if it's not a `BufferedReader` but if it is just use buffering?

Reply:
sure, sounds good :)